lib/BC/PcodeCFG.cpp

```diff
@@ -0,0 +1,166 @@
+  res.push_back(0);
+
+
+  for (size_t curr_index = 0; curr_index < this->linear_ops.size();
+       curr_index += 1) {
+    auto interproc = this->GetIntraProcTargets(curr_index);
+    res.insert(res.end(), interproc.begin(), interproc.end());
+  }
+  return res;
+}
+
+PcodeBlock PcodeCFGBuilder::BuildBlock(size_t start_ind, size_t next_start) {
+  std::vector<RemillPcodeOp> ops;
+
+  for (size_t i = start_ind; i < next_start; i++) {
+    ops.push_back(this->linear_ops[i]);
+  }
+
+  return PcodeBlock(start_ind, std::move(ops),
+                    this->GetBlockExitsForIndex(next_start - 1));
+}
+
+
+namespace {
+struct IntraProcTransferCollector {
+  std::vector<size_t> operator()(const InstrExit &ex) {
+    return std::vector<size_t>();
+  }
+
+  std::vector<size_t> operator()(const IntrainstructionIndex &ex) {
+    return std::vector<size_t>(1, ex.target_block_ind);
+  }
+
+
+  std::vector<size_t> operator()(const Exit &ex) {
+    return std::visit(*this, ex);
+  }
+
+  std::vector<size_t> operator()(const ConditionalExit &ex) {
+    std::vector<size_t> res = std::visit(*this, ex.true_branch);
+    std::vector<size_t> other = std::visit(*this, ex.false_branch);
+    res.insert(res.end(), other.begin(), other.end());
+    return res;
+  }
+};
+}  // namespace
+
+std::vector<size_t> PcodeCFGBuilder::GetIntraProcTargets(size_t index) {
+  std::variant<Exit, ConditionalExit> ex = this->GetBlockExitsForIndex(index);
+  return std::visit(IntraProcTransferCollector{}, ex);
+}
+
+std::variant<Exit, ConditionalExit>
+PcodeCFGBuilder::GetBlockExitsForIndex(size_t index) {
+  CHECK(index < this->linear_ops.size());
+  const auto &curr_op = this->linear_ops[index];
+
+  auto build_direct_target_exit = [](VarnodeData target,
+                                     size_t curr_ind) -> Exit {
+    if (isVarnodeInConstantSpace(target)) {
+      // need to treat as signed?
+      return IntrainstructionIndex{curr_ind + target.offset};
+    } else {
+      return InstrExit{};
+    }
+  };
+  switch (curr_op.op) {
+    case CPUI_BRANCH:
+    case CPUI_CALL: {
+      return Exit{build_direct_target_exit(curr_op.vars[0], index)};
+    }
+    case CPUI_CBRANCH: {
+
+
+      auto taken_exit = build_direct_target_exit(curr_op.vars[0], index);
+
+      Exit fallthrough_exit = InstrExit{};
+      // if we are not the last pcodeop then we have an intraproc fallthrough
+      if (index < this->linear_ops.size() - 1) {
+        fallthrough_exit = IntrainstructionIndex{index + 1};
+      }
+
+      return ConditionalExit{taken_exit, fallthrough_exit};
+    }
+    case CPUI_CALLIND:
+    case CPUI_BRANCHIND: {
+      return Exit{InstrExit{}};
+    }
+
+    default: {
+      return Exit{InstrExit{}};
+    }
+  }
+}
+
+
+PcodeBlock::PcodeBlock(size_t base_index)
+    : base_index(base_index),
+      ops(),
+      block_exit(Exit{InstrExit{}}) {}
+
+PcodeCFG PcodeCFGBuilder::Build() {
+
+  auto starts = this->GetBlockStarts();
+  std::set s(starts.begin(), starts.end());
```

Comment:
Unless I am missing something crucial:
```suggestion
  auto s = starts;
```

Reply:
The point of these two lines is to sort and de-duplicate the vector, ie. shove everything into a set and extract

Reply:
Yeah, add a comment, ideally also with reasoning why the order is not damaged (or why we don't care).

Reply:
[c095e26](https://github.com/lifting-bits/remill/pull/645/commits/c095e26f02a32457b6c65574c4f01e049b5fae6a)